libs/ui/design-system/src/lib/components/ButtonProgress/ButtonProgress.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+
+import { ButtonGroup as PrimerButtonGroup } from "@primer/react";
+import type { ButtonProps as PrimerButtonProps } from "@primer/react/deprecated";
+
+import "./ButtonProgress.scss";
+import { Button, EnumButtonStyle } from "../Button/Button";
+import classNames from "classnames";
+import { CircularProgress, circularProgressClasses } from "@mui/material";
+import type { CircularProgressProps } from "@mui/material";
+import { Box } from "@mui/material";
+import { margin } from "@mui/system";
+import {
+  EnumFlexDirection,
+  EnumGapSize,
+  EnumItemsAlign,
+  FlexItem,
+} from "../FlexItem/FlexItem";
+import { EnumFormStyle } from "../Form/Form";
+
+export interface Props
+  extends Pick<PrimerButtonProps, "className" | "children" | "onClick"> {
+  /** Progress percentage 0-100 */
+  progress: number;
+  yellowColorThreshold: number;
+  redColorThreshold: number;
+  leftValue: string;
+  rest?: any;
+}
+
+enum EnumButtonProgressStyle {
+  Primary = "primary",
+  Warning = "warning",
+  Danger = "danger",
+}
+
+const Pie = (props: CircularProgressProps) => {
+  return (
+    <Box
+      className={props.className}
+      sx={{
+        position: "relative",
+        display: "flex",
+      }}
+    >
+      <CircularProgress
+        {...props}
+        variant="determinate"
+        className=".amp-button-progress--background"
+        value={100}
+        thickness={3}
+        sx={{
+          color: "#686f8c",
+        }}
+      />
+      <CircularProgress
+        {...props}
+        variant="determinate"
+        value={props.value}
+        thickness={3}
+        sx={{
+          color: "inherit",
+          position: "absolute",
+        }}
+      />
+    </Box>
+  );
+};
+
+export const ButtonProgress = (props: Props) => {
+  const {
+    yellowColorThreshold,
+    redColorThreshold,
+    onClick,
+    leftValue,
+    className,
+    children,
+    rest,
+  } = props;
+  let { progress } = props;
+
+  if (progress < 0) {
```

Comment:
does progress can be negative number ? you can define the type and set default value
you can validate the prop by using ButtonProgress.propsType & ButtonProgress. defaultProps.
this will allow you to set default and pass valid number like:
progress: PropTypes.oneOf([...(new Array(10))].map((_, i) => i + 1))